### PR TITLE
Failing test for destroy not called on items in each

### PR DESCRIPTION
**Pull request by scottmessinger**
If I understand the problem correctly, I think this is a failing test for https://github.com/emberjs/ember.js/issues/18855

Let me know if there's anything I can do to help!

### CODE REVIEW

File: packages/@glimmer/integration-tests/test/updating-test.ts [hunk 1]
```diff
@@ -1893,8 +1893,11 @@ class UpdatingTest extends RenderTest {
     this.rerender({ list: ['initial', 'update'] });
     this.assertHTML(`<div>initial</div><div>update</div>`);
 
+    this.rerender({ list: ['initial'] });
+    assert.equal(destroyCount, 1, 'destroy was called for removed item');
+
     this.rerender({ list: [] });
-    assert.equal(destroyCount, 2, 'both list items were correctly destroyed');
+    assert.equal(destroyCount, 1, 'remaining list item was correctly destroyed');
```
Comment (rwjblue) [inline packages/@glimmer/integration-tests/test/updating-test.ts:1900]: I believe this should still be 2:

```suggestion
    assert.equal(destroyCount, 2, 'remaining list item was correctly destroyed');
```

Comment (scottmessinger) [reply packages/@glimmer/integration-tests/test/updating-test.ts]: You're totally right. Fixed and rebased.

**Comment by rwjblue** [comment]
OK, I think this should pass now if you rebase...

**Comment by scottmessinger** [comment]
@rwjblue Rebased and tests now pass!

**Pull request merged**